From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 3 Jul 2024 14:47:41 -0500
Subject: [PATCH] Solus: Use containerd and other defaults

Changes:
- Source /etc/sysconfig/buildkitd if it exists
- Allow users to use buildkit if they are in the buildkit group
- Use containerd runtime

---
 examples/systemd/system/buildkit.service | 11 ++++++++---
 examples/systemd/system/buildkit.socket  |  2 ++
 2 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/examples/systemd/system/buildkit.service b/examples/systemd/system/buildkit.service
index 46052401..f90764f2 100644
--- a/examples/systemd/system/buildkit.service
+++ b/examples/systemd/system/buildkit.service
@@ -1,12 +1,17 @@
 [Unit]
-Description=BuildKit
+Description=BuildKit Daemon
 Requires=buildkit.socket
-After=buildkit.socket
+After=buildkit.socket containerd.service
+Wants=containerd.service
 Documentation=https://github.com/moby/buildkit
 
 [Service]
 Type=notify
-ExecStart=/usr/local/bin/buildkitd --addr fd://
+EnvironmentFile=-/etc/sysconfig/buildkitd
+ExecStart=/usr/bin/buildkitd --addr fd:// \
+          --oci-worker=false \
+          --containerd-worker=true \
+          $BUILDKITD_OPTIONS
 
 [Install]
 WantedBy=multi-user.target
diff --git a/examples/systemd/system/buildkit.socket b/examples/systemd/system/buildkit.socket
index 3f9a35e9..776b2376 100644
--- a/examples/systemd/system/buildkit.socket
+++ b/examples/systemd/system/buildkit.socket
@@ -5,6 +5,8 @@ Documentation=https://github.com/moby/buildkit
 [Socket]
 ListenStream=%t/buildkit/buildkitd.sock
 SocketMode=0660
+SocketUser=root
+SocketGroup=buildkit
 
 [Install]
 WantedBy=sockets.target
